src/Libraries/Microsoft.Extensions.AI/Functions/AIFunctionException.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI.Functions;
+
+/// <summary>
+/// Defines an exception raised in an <see cref="AIFunction"/> invocation containing a message intended for the model detailing the reason for the failure.
+/// </summary>
+#pragma warning disable CA1032 // Implement standard exception constructors
+public sealed class AIFunctionException : Exception
```

Comment:
With exception types it is generally a good idea to keep them unsealed.
```suggestion
public class AIFunctionException : Exception
```